filters/tee/tee.go

```diff
@@ -46,8 +45,8 @@ type Options struct {
 	// Timeout specifies a time limit for requests made by tee filter.
 	Timeout time.Duration
 
-	// Tracer is the opentracing tracer to use in the client
-	Tracer opentracing.Tracer
+	// Tracer is the open telemetry tracer to use in the client
+	Tracer trace.Tracer
```

Comment:
sometimes I see `Tracer` and others `OtelTracer`, is this intended?

Reply:
Yeah, in some places I used `OtelTracer` to differentiate from `Tracer` and `OtTracer`, the former is only OT if we need to be backward compatible and the later is kept in places we need access to both tracers and I wanted to make it clear that it is one or the other. Marjority of cases it should be just `tracer` or `Tracer`.

Reply:
Also, when we discontinue OpenTracing,  which will probably take a while, we can have all of them as just Tracer/tracer.